http/longpoll/longpoll.go

```diff
@@ -0,0 +1,75 @@
+package longpoll
+
+import (
+	"context"
+	"time"
+)
+
+// LongPollFunc should return true if long polling did
+// resolve, false otherwise. If error is returned, the
+// long polling will be canceled
+type LongPollFunc func() (bool, error)
```

Comment:
Do we want to add a `context.Context` to the `LongPollFunc` so that it can explicitly react to context "events" (cancelation etc.) of the "parent" context, i.e, the one given by `Until(ctx context.Context, ...)`.

The caller then could still choose to ignore the context in the function, but would have the ability to use it without explicitly passing it into the function body when the function is being defined.

Just a thought of a possibly "problematic" scenario would be, the a `LongPollFunc` gets created "somewhere" and spawns a goroutine that gets terminated by a context with deadline or something similar. If we now pass by accident or whatever reasons two different contexts into `Until` we might end up with strange behaviour such as, the long polling randomly failing after 10 seconds, although the `MinWaitTime` is like 60 seconds.
We would see the context cancellation error, yes, but we would need to track down which context actually caused the cancellation. And when the definition of a `LongPollFunc` and the call to `Until` are in different scopes this might be hard to keep track off.

Reply:
I like the idea, will add.

Reply:
I second the idea of passing the context to the `LongPollFunc`.